### ISSUE TITLE
Airspeed indicator is broken!

### ISSUE THREAD

**Issue by gilbertohasnofb** [Juanvvc/c172p-detailed#773]
Reported by mifi in the forum:

> I noticed the airspeed indicator in the C172 detailed does not work in the version of flightgear I compiled today. The C172 detailed was installed via the new launcher.
> The indicated airspeed stays firmly at zero, even when flying.
> 
> Flightgear version shows 2016.3.0.
> Is it just me?

Indeed I can reproduce it. I found that our `c172p-set.xml` file contains the following code:

```
        <airspeed-indicator>
            <serviceable type="bool">false</serviceable> <!-- since pitot tube is on for the first time one runs this plane -->
        </airspeed-indicator>
```

Are we still using the `instrumentation/airspeed-indicator/serviceable` property to break the AI? Because that's the only mention of it I could find in the whole project using `grep -R "instrumentation/airspeed-indicator/serviceable" *.xml`.

**Comment by wlbragg** [comment]
So if you remove the pitot cover this is set to true?
If pitot condition is persistent data, why does this break it?

**Comment by gilbertohasnofb** [comment]
I think the problem is that in the past we used that `serviceable` to break the AI, but now we use some other method. So now regardless of the cover, the `serviceable` should be always `true`.

**Comment by wlbragg** [comment]
>Because that's the only mention of it I could find in the whole project 

Meaning we never change this when the cover is removed?

**Comment by gilbertohasnofb** [comment]
> Meaning we never change this when the cover is removed?

Yep! In the past it used to, but I think there was some huge overhaul of the original code of the tiedowns, pitot cover, etc. some time ago, and since then the techniques used changed. But it seems this little detail went unnoticed

**Comment by wlbragg** [comment]
Is there any reason we might want to keep the airspeed-indicator/serviceable parameter but make it "true". Does it need to exist for any other reason like "failure" code?

**Comment by onox** [comment]
I barely know what this bug is about, but is this bug one those that are solved by throwing away some temporary files in `.fgfs/`?

**Comment by gilbertohasnofb** [comment]
> I barely know what this bug is about, but is this bug one of those that are solved by throwing away some temporary files in .fgfs/?

I tried that, for me it doesn't solve. The problem is kind of simple: if the property `instrumentation/airspeed-indicator/serviceable` is set to `false`, then the AI fails. I used this property when I first implemented the pitot cover. Now, as far as I could tell, our -set file sets it to `false` but no other code sets it back to `true`, so regardless of what I do it's broken! So even if one deletes the temp fie in the `.fgfs` folder, the -set file will set it to `false` and break it.

**Comment by wlbragg** [comment]
We have another possible issue
https://forum.flightgear.org/viewtopic.php?f=25&t=29694
Aircraft won't start. I'm trying to get a bit more info though before moving it here.

**Comment by gilbertohasnofb** [comment]
> We have another possible issue

When the guy gives more info about it I will try to reproduce. Right now I can start the aircraft without any problems even in very cold weather (tested now at -10 oC)

**Comment by wlbragg** [comment]
>13700ft MSL... the craft's ceiling is like 13000ft up to 13500ft depending on where you look

The airport is too high.

**Comment by wkitty42** [comment]
On 05/30/2016 02:53 PM, wlbragg wrote:
> We have another possible issue
> https://forum.flightgear.org/viewtopic.php?f=25&t=29694
> Aircraft won't start. I'm trying to get a bit more info though before moving it
> here.

i'm pretty sure it is altitude related... the airport in question is at 13700ft 
MSL and our craft has a service ceiling of 13000ft to 13500ft... he mentioned 
the piper working which has a ceiling of 14300ft...

i have flown our craft to 13500ft and had some winds push me up to 13700ft but i 
never tried turning off the engine and restarting it at that altitude...

**Comment by gilbertohasnofb** [comment]
OMG, the airport is at 13.700 and the temp is -10, no wonder that poor engine can't handle it !

**Comment by wlbragg** [comment]
Anyone know where the code that limits the ceiling is?

**Comment by gilbertohasnofb** [comment]
This is a guess, but I don't think it's that simple as a limitation. I think it is due to a bunch of factors (density, pressure, humidity, etc.) which the engine probably takes into account.

**Comment by wkitty42** [comment]
yup, that's my impression, too... you can fly up there but apparently you can't start up there... when i was testing that scenario, the autostart just kept turning over and over until i finally hit the 's' to stop it...

**Comment by wlbragg** [comment]
I guess the only thing that would concern me is if we aren't portraying this behavior accurately enough.

**Comment by onox** [comment]
`serviceable` properties of ASI and AI are true when I start FG with the c172p.

**Comment by gilbertohasnofb** [comment]
And can you confirm your `c172p-set.xml` file has it being set to `false` on line 602? If so, then how on Earth is it changing to `true` when the property isn't mentioned on any other file of our project?!

**Comment by onox** [comment]
What kind of mixture does he use? Trying to start the engine at 13700 ft is a bit insane

**Comment by onox** [comment]
I'm at the latest commit in `master`, so yes, it is set to false in the file. Maybe it's done by some C++ in FG itself.

**Comment by gilbertohasnofb** [comment]
> What kind of mixture does he use?

And for the note, autostart always set mixture to full rich

> I'm at the latest commit in master, so yes, it is set to false in the file. Maybe it's done by some C++ in FG itself.

Anyway, I would recommend removing that property from the set file since we are not using it anymore, and since it fixes the airspeed indicator at least on my side. @wlbragg do you also see the ASI problem? And does it go away on branch `bug-773`?

**Comment by wlbragg** [comment]
Hum, still not working in branch bug-733?

**Comment by wlbragg** [comment]
Not only that but if I put it back in and set it to true it works. So unless someone wants to track down exactly what it is for and doing, I would suggest leaving it in and setting it to true (we would need to test if the pitot cover still works. If not then it would need to be set to false when the cover was on.

**Comment by gilbertohasnofb** [comment]
> Hum, still not working in branch bug-733?

Can you please check the value of `instrumentation/airspeed-indicator/serviceable` in the property tree? And did you try it only once or twice? It might be necessary to load the sim twice to fix it on that branch because the value of that property was saved on exit, so the first time you load it the sim will load it to be `false`. But then next time you quit that property won't be saved at all in the `.fgfs`, so this time it's not being set to `false` by the -set.xml file nor is it being set to false by the data file in `.fgfs`.

Can you please try that before we setting it to `true`, as I find that solution a bit inelegant?

Here is how to test it now:

- first, load the plane from the `master` branch, so that your data file inside `.fgfs` saves that property back (so that we can check if the bug disappears for you)
- then load the sim using the branch `bug-773` and check the IAS. It probably won't be working. Quit the sim via the `ESC` key so it overwrites that data file
- load the sim again using the branch `bug-773`

**Comment by wkitty42** [comment]
FWIW: serviceable being true seems to be the key... i was going to do the above but forgot to load, exit, reload... but i went ahead and looked at that property... it was false and ctrl-clicking on it set it true and immediately the device started working... i'm on the master version of the craft... after this flight is completed, i'll try it again and see what happens...

note: at this time, during this flight, i have looked at the .fgfs/aircraft-data/c172p.xml file and did find the airspeed-indicator property set to false in there... in my case, it is the very first entry in the instrumentation section... i'm going to leave it alone and see what happens at the end of this flight now that i have set it to true in the sim... if i am reading correctly, it should not be saved between sessions any more...

**Comment by wkitty42** [comment]
ok, following up on my post above... i had set the instrumentation/airspeed-indicator/serviceable property to true while flying the master c172p... after i landed and exited the sim, the property was (still?) written to the .fgfs/aircraft-data/c172p.xml file with "true"... i loaded the sim again and the airspeed-indicator was working properly...

i'm starting to wonder how and when the property was ever set to false... in any case, as noted above, once i set it to true, it worked and has in the few times that i've flown afterward...

**Comment by gilbertohasnofb** [comment]
@wkitty42 was this done on `master` branch? It's strange that the -set file wouldn't override the saved status of that property: that is, I imagined it would force it to false next time you loaded the sim.

@all But personally, I think the most obvious way of solving this is really what I did on the PR https://github.com/Juanvvc/c172p-detailed/pull/775 since there is no reason for us to neither set nor save this property. It's causing problems, so let's just remove it. If @wlbragg confirms that the steps I posted above do solve his problems, then I'd say we should merge that PR and let the devs know ASAP about the need of a new merge.

**Comment by wlbragg** [comment]
@gilbertohasnofb I'm not sure I did exactly what you requested, but I can tell you this. With out the  serviceable property being defined in the set, I now have it show up in debug but it is set to true. I set it to false manually, closed the sim, restarted the sim and it was set back to true. The odd thing being I have that property at all. Bottom line is it seems to work. I tested the pitot cover and it still worked correctly.

**Comment by gilbertohasnofb** [comment]
So in that case I think we are ready to merge that PR. @onox do you have anything for/against it? 

@wkitty42 did you test the branch `bug-773` and does it fix the issue for you? You can try following these steps: https://github.com/Juanvvc/c172p-detailed/issues/773#issuecomment-222667823 

I just want to confirm this fixes the issues to everyone, since it seemed that each of us were experiencing the bug in a slightly different way

**Comment by wkitty42** [comment]
On 05/31/2016 03:27 PM, Gilberto Agostinho wrote:
> @wkitty42 was this done on |master| branch? It's strange that the -set file
> wouldn't override the saved status of that property: that is, I imagined it
> would force it to false next time you loaded the sim.

yes, thus my use of "master" ;)

> @all But personally, I think the most obvious way of solving this is really
> what I did on the PR #775 since there is no reason for us to neither set nor
> save this property. It's causing problems, so let's just remove it. If
> @wlbragg confirms that the steps I posted above do solve his problems, then
> I'd say we should merge that PR and let the devs know ASAP about the need of
> a new merge.

it doesn't really matter to me... if we don't need to save this property then 
let's remove it from the aircraft-data file...

i'm assuming that the pitot cover still makes the device read zero if we forget 
to remove it? it does when i test it in master by rolling down the runway with 
it on... then when i stop and remove it and apply throttle again, it works... 
the problem seems to be that it simply got set to false somehow somewhen...

the very first time the craft is loaded, the property IS set to FALSE because 
there's no existing saved data in the .fgfs/aircraft-data/c172p.xml file... the 
default load for the craft does not have the pitot tube cover in place even 
though the comment in the -set file for the property says that the pitot cover 
is in place... without digging through the repo history, it looks like our 
defaults have changed for a newly loaded craft and that's when this item was 
missed... how and when it was ever set to true must have been or planned to be 
in the pitot cover code for when the cover was removed...

/lack of c0ffee

**Comment by gilbertohasnofb** [comment]
Ok, thanks a lot!

So what about this: let's force the property to `true` via the -set.xml file, but let's remove it from the aircraft-data file. This will surely solve any possible problems and will not conflict with the pitot cover. It will also override any saved data from old aircraft-data files that some users might have on their computers, thus forcing the bug to be solved. What do you all think? I can update that PR if you are all agree.

**Comment by wkitty42** [comment]
On 05/31/2016 06:47 PM, Gilberto Agostinho wrote:
> So in that case I think we are ready to merge that PR. @onox do you have
> anything for/against it?
>
> @wkitty42 did you test the branch |bug-773| and does it fix the issue for
> you? You can try following these steps: #773 (comment)

i have not tested against any branch other than master... TBH, i'm not sure how 
i would do that... is it simply a matter of checking it out, running the sim and 
then remembering to check out master when done? i guess the 
aircraft-data/c172p.xml file should be deleted each time a different branch is 
tested??

> I just want to confirm this fixes the issues to everyone, since it seemed
> that each of us were experiencing the bug in a slightly different way

i didn't see the bug at all until i deleted my .fgfs/aircraft-data/c172p.xml 
file to start with a clean slate... before than it worked fine...

**Comment by wkitty42** [comment]
On 06/01/2016 10:13 AM, Gilberto Agostinho wrote:
> Ok, thanks a lot!
>
> So what about this: let's force the property to |true| via the -set.xml file,
> but let's remove it from the aircraft-data file. This will surely solve any
> possible problems and will not conflict with the pitot cover. It will also
> override any saved data from old aircraft-data files that some users might have
> on their computers, thus forcing the bug to be solved. What do you all think? I
> can update that PR if you are all agree.

that sounds fine to me...

**Comment by wlbragg** [comment]
>So what about this: let's force the property to |true| via the -set.xml file,

Why do we have to do this? Is it because on a clean install it defaults to "false"? Because as I said, mine now defaults to true no mater how I leave the state of serviceable when I exit the sim.

**Comment by gilbertohasnofb** [comment]
IMO it's not necessary, but if someone has the aircraft-data file then for the first time it will load the `false` value (since it was saved from the previous version). Then IF the person quits the sim correctly (not by clicking on the x), that file will be overwritten and there will be no value for that property, and FG default is obviously `true`. So if we set it to `true`, we avoid that an user with a non clean install will have problems for on the first load in the sim, and potentially on all others if he/she closes it via the x in the window. 

So it's up to you all, if we want to simply remove the property from the -set.xml file and and from the aircraft-data, then the PR is ready. If we want to force it to `true` just in case, then I need to push another commit.

**Comment by wlbragg** [comment]
Hum, I see the predicament. Well the only way to insure the problem doesn't bite someone is to default it to true in the set or on init with a if false = true.

Or we can just make a statement on the c172p thread explaining the potential problem and solution.

**Comment by gilbertohasnofb** [comment]
@onox any opinions on this at all? Can I go ahead and set it to `true` and we are good to go?

@all I _really_ think we need to act as fast as possible, 2016.2.2 is around the corner and we can't just ship it with a broken airplane!

**Comment by wlbragg** [comment]
Just do it, we can always remove it later after a few FG versions go by. I'll still post something on the forum explaining what's up.

**Comment by wlbragg** [comment]
Pending @onox's reply. :smile:

**Comment by gilbertohasnofb** [comment]
Ok, so as soon as @onox reply (and if it's a green light) then I will go ahead and push the commit changing forcing that property to `true`

And in the forum you won't need to go into much details, since this solution will surely fix it for everyone. So you can just say we found the bug and solved it. Though that's up to you.

**Comment by gilbertohasnofb** [comment]
@onox? Are you there?

**Comment by onox** [comment]
> @onox? Are you there?

Yes, I'm still here, but I'm a bit busy with non-FG OSS projects.

**Comment by gilbertohasnofb** [comment]
> Yes, I'm still here, but I'm a bit busy with non-FG OSS projects.

@onox I totally understand that, but if you have a chance to reply we are just waiting for a single opinion on this issue: should we set the property to `true` to avoid any issues that could be caused by the `.fgfs/aircraft-data/c172p.xml` or should we remove any mention of it from the set file? I think both @wlbragg and I prefer the first, as it avoids possible problems for all users under the cost of a single (albeit inelegant) line of code.

If you agree with us, I would then push a commit setting that property to `true` and then we could merge that PR, and also start to talk about a new merge into FGDATA.

**Comment by onox** [comment]
i don't have any opinion about the issue, so you can add that single line if you want.

**Comment by gilbertohasnofb** [comment]
:+1:
